checkstyle inconsistent with prettier

```
[ant:checkstyle] [ERROR] Merge.java:40:80: Only one statement per line allowed. [OneStatementPerLine]
```

```
npx prettier --version
3.1.1
2.5.0
```

```yml
plugins:
  - prettier-plugin-java
  - prettier-plugin-properties
  - prettier-plugin-sh
  - prettier-plugin-toml
  - "@prettier/plugin-xml"
```

checkstyle, I copied pasta-ed from https://github.com/jhipster/prettier-java/commit/deda6ca37214420c37b5981ca23119ef49743620#diff-6c435908fe54a62733ee357cc3ef828f242045ba31fa59d6d3f763b5352d694e which is just a link to the version in master at the time of this writing

**Input:**

```java
        if (first == null) return second;
        else if (second == null) return first;
```

**Output:**

```java
        if (first == null) return second; else if (second == null) return first;
```

**Expected behavior:**

no error, results should be consistent, however I'm not certain if prettier is creating the wrong output or if checkstyle is configured wrong. I'm kind of wondering if prettier can/should be adding the braces here in the first place.

--- a/packages/prettier-plugin-java/src/printers/blocks-and-statements.ts
+++ b/packages/prettier-plugin-java/src/printers/blocks-and-statements.ts
@@ -205,7 +205,9 @@ export class BlocksAndStatementPrettierVisitor extends BaseCstPrettierPrinter {
 
       const elseOnSameLine =
         hasTrailingLineComments(ctx.statement[0]) ||
-        hasLeadingLineComments(ctx.Else[0])
+        hasLeadingLineComments(ctx.Else[0]) ||
+        !ctx.statement[0].children.statementWithoutTrailingSubstatement?.[0]
+          .children.block
           ? hardline
           : " ";
 

--- a/packages/prettier-plugin-java/test/unit-test/empty_statement/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/empty_statement/_output.java
@@ -37,7 +37,8 @@ public class EmptyStament {
   }
 
   public void ifElseWithEmptyStatements() {
-    if (test); else {
+    if (test);
+    else {
       System.out.println("one");
     }
 
@@ -45,15 +46,18 @@ public class EmptyStament {
       System.out.println("two");
     } else;
 
-    if (test); else;
+    if (test);
+    else;
   }
 
   public void ifElseWithEmptyStatementsWithComments() {
-    if (test) /*test*/; else {
+    if (test) /*test*/;
+    else {
       System.out.println("one");
     }
 
-    if (test); /*test*/else {
+    if (test);
+    /*test*/else {
       System.out.println("one");
     }
 
@@ -65,9 +69,11 @@ public class EmptyStament {
       System.out.println("two");
     } else;/*test*/
 
-    if (test); /*test*/else;/*test*/
+    if (test);
+    /*test*/else;/*test*/
 
-    if (test) /*test*/; else /*test*/;
+    if (test) /*test*/;
+    else /*test*/;
   }
 
   public void simpleWhileWithEmptyStatement(boolean one) {

--- a/packages/prettier-plugin-java/test/unit-test/sealed/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/sealed/_output.java
@@ -46,11 +46,9 @@ public sealed interface Shape permits Circle, Rectangle, Triangle, Unicorn {
   }
 
   default String areaMessage() {
-    if (this instanceof Circle) return "Circle: " + area(); else if (
-      this instanceof Rectangle
-    ) return "Rectangle: " + area(); else if (
-      this instanceof RightTriangle
-    ) return "Triangle: " + area();
+    if (this instanceof Circle) return "Circle: " + area();
+    else if (this instanceof Rectangle) return "Rectangle: " + area();
+    else if (this instanceof RightTriangle) return "Triangle: " + area();
     // :(
     throw new IllegalArgumentException();
   }

